docs/en/operation/Troubleshooting.md

```diff
@@ -269,6 +269,67 @@ are
 - `ADVANCED`: samples and tracks recent accesses, higher overhead
 - `PARANOID`: tracks for leaks on every resource allocation, highest overhead. 
 
+## Master Internal Monitoring
+
+Alluxio master periodically checks the master process resource usage. This interval is controlled by
+`alluxio.master.throttle.heartbeat.interval` (default 3 seconds). In the check, a point in time(PIT) resource usage is captured,
+and a certain number of PITs (`alluxio.master.throttle.observed.pit.number`, default 3) are recorded. The PIT includes the following information.
+
+```console
+directMemUsed=5268082, heapMax=59846950912, heapUsed=53165684872, cpuLoad=0.4453061982287778, pitTotalJVMPauseTimeMS=190107, totalJVMPauseTimeMS=0, rpcQueueSize=0, pitTimeMS=1665995384998}
+```
+
+- `directMemUsed`: memory allocated by ByteBuffer.allocateDirect
+- `heapMax` : the allowed max heap size
+- `heapUsed` : the heap memory used
+- `cpuLoad` : the cpu load
+- `pitTotalJVMPauseTimeMS` : aggregated total JVM pause time
+- `totalJVMPauseTimeMS` : the JVM pause time since last PIT
+- `rpcQueueSize` : the rpc queue size
+- `pitTimeMS` : the point in time
+
+The aggregated server indicators are the certain number of continuous PITs, this one is generated in a sliding window. The alluxio
+master has a derived indicator `Master.system.status` that is based on the heuristic algorithm.
+
+```console
+    "Master.system.status" : {
+      "value" : "STRESSED"
+    }
+```
+
+The possible value are: 
+- IDLE
+- ACTIVE
+- STRESSED
+- OVERLOADED
+
+The system.status is mainly decided by the JVM pause time and the free heap memory. Usually the status transition is 
+`IDLE` <---> `ACTIVE` <---> `STRESSED` <---> `OVERLOADED`
+- If the JVM pause time is longer than `alluxio.master.throttle.overloaded.heap.gc.time`, the system.status is directly set to `OVERLOADED`.
+- If the used heap memory is greater than the low used heap memory boundary threshold, the system.status is `DEESCALATED`.
+- If the used heap memory is less than the up used heap memory boundary threshold, the system.status is `UNCHANGE`
+- If the aggregated used heap memory is greater than the up used heap memory boundary threshold, the sytem.status is `ESCALATED`.
```

Comment:
These are design details, which belong in internal design docs.
They can evolve quite quickly as we find new metrics to add to the algorithm, and the specific rules applied to determine the state transitions are subject to change, too.